x-pack/plugins/cases/public/components/all_cases/table_filter_config/use_filter_config.tsx

```diff
@@ -0,0 +1,144 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useEffect, useState } from 'react';
+import useLocalStorage from 'react-use/lib/useLocalStorage';
+import { LOCAL_STORAGE_KEYS } from '../../../../common/constants';
+import type { FilterChangeHandler, FilterConfig, FilterConfigState } from './types';
+import { useCustomFieldsFilterConfig } from './use_custom_fields_filter_config';
+import { MoreFiltersSelectable } from './more_filters_selectable';
+import { useCasesContext } from '../../cases_context/use_cases_context';
+
+const mergeSystemAndCustomFieldConfigs = ({
+  systemFilterConfig,
+  customFieldsFilterConfig,
+}: {
+  systemFilterConfig: FilterConfig[];
+  customFieldsFilterConfig: FilterConfig[];
+}) => {
+  const newFilterConfig = new Map(
+    [...systemFilterConfig, ...customFieldsFilterConfig]
+      .filter((filter) => filter.isAvailable)
+      .map((filter) => [filter.key, filter])
+  );
+
+  return newFilterConfig;
+};
+
+export const useFilterConfig = ({
+  systemFilterConfig,
+  onFilterOptionChange,
+}: {
+  systemFilterConfig: FilterConfig[];
+  onFilterOptionChange: FilterChangeHandler;
+}) => {
+  const { appId } = useCasesContext();
+  const { customFieldsFilterConfig } = useCustomFieldsFilterConfig({ onFilterOptionChange });
+  const [filterConfigs, setFilterConfigs] = useState<Map<string, FilterConfig>>(
+    () => new Map([...systemFilterConfig].map((filter) => [filter.key, filter]))
+  );
+  const [activeByFilterKey, setActiveByFilterKey] = useLocalStorage<FilterConfigState[]>(
+    `${appId}.${LOCAL_STORAGE_KEYS.casesTableFiltersConfig}`,
+    []
+  );
+
+  useEffect(() => {
+    const newFilterConfig = mergeSystemAndCustomFieldConfigs({
+      systemFilterConfig,
+      customFieldsFilterConfig,
+    });
+
+    filterConfigs.forEach((filter) => {
+      if (!newFilterConfig.has(filter.key)) {
+        filter.deactivate();
```

Comment:
we decided to try another approach. Instead of calling one by one, each filter would return its part so we can merge everything together and do just one call